### PR TITLE
[clang][analyzer] Ignore try-statements in UnreachableCode checker

### LINKED ISSUE #90162: [clang][static analyzer]alpha.deadcode.UnreachableCode false positive on cxxTryStmt
```cpp
  class BaseException {};
  
  class DerivedException : public BaseException {};
  
  void foo();
  
  void f4() {
    try {
      foo();
    } catch (BaseException &b) {
    } catch (DerivedException &d) {
    }
  }
```

When I run ```clang-tidy --allow-enabling-analyzer-alpha-checkers -checks=-*,clang-analyzer-alpha.deadcode.UnreachableCode test.cpp```, the result like as following:

```bash
test.cpp:8:3: warning: This statement is never executed [clang-analyzer-alpha.deadcode.UnreachableCode]
    8 |   try {
      |   ^~~~~
    9 |     foo();
      |     ~~~~~~
   10 |   } catch (BaseException &b) {
      |   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   11 |   } catch (DerivedException &d) {
      |   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   12 |   }
      |   ~
test.cpp:8:3: note: This statement is never executed
    8 |   try {
      |   ^~~~~
    9 |     foo();
      |     ~~~~~~
   10 |   } catch (BaseException &b) {
      |   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   11 |   } catch (DerivedException &d) {
      |   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   12 |   }
      |   ~

```

I think this is false positive. If not , I am not clear about reason.

### PR DESCRIPTION
**Pull request by sookach**
Fixes #90162. Simplest approach I could come up with was to skip cxxtry statements. Let me know if you have any suggestions. Thanks!

### CODE REVIEW

File: clang/lib/StaticAnalyzer/Checkers/UnreachableCodeChecker.cpp [hunk 1]
```diff
@@ -159,8 +159,10 @@ void UnreachableCodeChecker::checkEndAnalysis(ExplodedGraph &G,
       SL = DL.asLocation();
       if (SR.isInvalid() || !SL.isValid())
         continue;
-    }
-    else
+
+      if (isa<CXXTryStmt>(S))
```
Comment (soukatch) [inline clang/lib/StaticAnalyzer/Checkers/UnreachableCodeChecker.cpp:163]: everything above this point is just clang-format changes

**Comment by soukatch** [comment]
@steakhal I believe you're the best person to tag :).

File: clang/lib/StaticAnalyzer/Checkers/UnreachableCodeChecker.cpp (hunk 1 above)
Comment (soukatch) [inline clang/lib/StaticAnalyzer/Checkers/UnreachableCodeChecker.cpp:163]: everything besides this if statement is just a clang-format change

**Review by steakhal — CHANGES_REQUESTED** [review]
Please remove the formatting changes. (Keep only the relevant lines)
Add a test demonstrating that this fixes a false-positive.

To find which file you need to add your test, I'd recommend braking something inside the checker to see which tests break. Then pick the file that seems to be the most relevant.

**Comment by soukatch** [comment]
> To find which file you need to add your test,

I created a new file altogether since I don't think any of the current one would fit the test.

File: clang/test/Analysis/unreachable-code-exceptions.cpp [hunk 2]
```diff
@@ -0,0 +1,12 @@
+// RUN: %clang_analyze_cc1 -verify %s -fcxx-exceptions -fexceptions -analyzer-checker=core -analyzer-checker=alpha.deadcode.UnreachableCode
```
Comment (steakhal) [inline clang/test/Analysis/unreachable-code-exceptions.cpp:1]:
```suggestion
// RUN: %clang_analyze_cc1 -verify %s -fcxx-exceptions -fexceptions -analyzer-checker=core,alpha.deadcode.UnreachableCode
```

File: clang/test/Analysis/unreachable-code-exceptions.cpp [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+// expected-no-diagnostics
+
+void foo();
+
+void f4() {
+  try {
+    foo();
+  } catch (int) {
+  }
+}
```
Comment (steakhal) [inline clang/test/Analysis/unreachable-code-exceptions.cpp:12]:
```suggestion
void fp_90162() {
  try { // no-warning: The TryStmt shouldn't be unreachable.
    foo();
  } catch (int) {
    foo(); // We assume that catch handlers are reachable.
  }
}

```

**Review by steakhal — APPROVED** [review]

**Comment by soukatch** [comment]
Thanks for the review!

**Pull request merged**